### PR TITLE
DM-21003: Take another look at pex_exceptions hierarchy

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: python/lsst/pex/exceptions/wrappers.py [hunk 1]
```diff
@@ -81,7 +82,7 @@ def __str__(self):
 
 
 @register
-class LogicError(Exception):
+class LogicError(Exception, builtins.Exception):
```
Comment (ktlim) [inline python/lsst/pex/exceptions/wrappers.py]: Isn't `builtins.Exception` already a base of `Exception`?

File: python/lsst/pex/exceptions/wrappers.py [hunk 2]
```diff
@@ -116,27 +117,27 @@ class RangeError(RuntimeError):
 
 
 @register
-class OverflowError(RuntimeError, OverflowError):
+class OverflowError(Exception, builtins.OverflowError):
```
Comment (ktlim) [inline python/lsst/pex/exceptions/wrappers.py]: I thought the intent here was to replicate the C++ inheritance hierarchy on the left and the Python one on the right.  In C++, `overflow_error` does derive from `runtime_error`.

**Review by ktlim — CHANGES_REQUESTED** [review]
I'm OK with the additions of `builtins`, which I think are clarifying and should not change things.  I am not so OK with the changing of `(lsst.pex.exceptions.)RuntimeError` to `(lsst.pex.exceptions.)Exception` as a base class for several errors unless a clear usage problem is given as a test case (not just a "did the code do what I said" test case as is currently present).

File: python/lsst/pex/exceptions/wrappers.py (hunk 2 above)
Comment (timj) [reply python/lsst/pex/exceptions/wrappers.py]: Yes, now I'm not entirely sure what @brianv0 's problem really is. This ticket only really changes `UnderflowError`, `OverFlowError` and `IoError`, yet I thought @brianv0 was seeing a problem with `NotFoundError`, which is only changing here with the addition of an explicit `builtins`.  I'll repush this ticket without the `RuntimeError` changes and ask him to test it.

**Review by ktlim — APPROVED** [review]

**Pull request merged**